amlip_docs/rst/demo/workload_distribution.rst

```diff
@@ -0,0 +1,137 @@
+.. include:: ../exports/alias.include
+
+.. _demo_workload_distribution:
+
+##########################
+Workload Distribution Demo
+##########################
+
+This demonstrator shows how to implement 2 kind of nodes:
+:ref:`user_manual_nodes_computing` and :ref:`user_manual_nodes_main`.
+With these 2 nodes implemented, the user can deploy as many nodes of each kind as desired and check the
+behavior of a simulated |amlip| network running.
+They are implemented one in Python and one in C++ to demonstrate as well how to instantiate each kind of node
+with different :term:`APIs <API>`, and to probe the communication between the 2 implementations.
+
+Simulation
+==========
+
+AML Mock
+--------
+
+For this demo the actual :term:`AML` Engine is not provided, and it is mocked.
+This *Mock* simulates a difficult calculation by converting a string to uppercase and randomly waiting between
+1 and 5 seconds in doing so.
+
+Main Node
+---------
+
+This node simulates a :ref:`user_manual_nodes_main`.
+It does not use real *AML Jobs*, but strings.
+It is implemented in |python| using :code:`amlip_py` API.
+There are 2 different ways to run it, an automatic one and a manual one:
+
+Automatic version
+^^^^^^^^^^^^^^^^^
+
+In this version, the python executable expects input arguments.
+For each argument, it will convert it to a string (:code:`str`) and send it as a *Job*.
+Once the arguments run out, it will finish execution and destroy the Node.
+
+Manual version
+^^^^^^^^^^^^^^
+
+In this version the python program expects to receive keyboard input.
+For each keyboard input received, it will convert it to a string (:code:`str`) and send it as a *Job*.
+When empty string given, it will finish execution and destroy the Node.
+
+Computing Node
+--------------
+
+This node simulates a :ref:`user_manual_nodes_computing`.
+It does not use real *AML Jobs*, but strings.
+It does not have a real *AML Engine* but instead the calculation is an *upper-case* conversion of the string received.
+It is implemented in |cpp| using :code:`amlip_cpp` API.
+
+To run it, one integer argument is required. This will be the number of jobs this Node will answer
+before finish execution and be destroyed.
```

Comment:
```suggestion
To run it, one integer argument is required. This will be the number of jobs this Node will answer to
before finishing its execution and being destroyed.
```